src/charm.py

```diff
@@ -37,53 +34,41 @@ def _on_collect_unit_status(self, event: CollectStatusEvent):
         Args:
             event: CollectStatusEvent
         """
-        if invalid_configs := self._get_invalid_configs():
+        if not self._relation_created(CORE_GNB_RELATION_NAME):
             event.add_status(
-                BlockedStatus(f"The following configurations are not valid: {invalid_configs}")
+                BlockedStatus(f"Waiting for {CORE_GNB_RELATION_NAME} relation to be created")
             )
-            logger.info("The following configurations are not valid: %s", invalid_configs)
+            logger.info("Waiting for %s relation to be created", CORE_GNB_RELATION_NAME)
             return
-        event.add_status(ActiveStatus())
+        if not self._core_gnb_requirer.tac or not self._core_gnb_requirer.plmns:
+            event.add_status(WaitingStatus("Waiting for TAC and PLMNs configuration"))
+            return
+        event.add_status(
+            ActiveStatus(
+                f"PLMNs: {','.join([str(plmn.asdict()) for plmn in self._core_gnb_requirer.plmns])}, "  # noqa: E501
+                f"TAC: {self._core_gnb_requirer.tac}"
+            )
+        )
 
-    def _on_config_changed_or_relation_joined(self, event: EventBase):
-        """Publish the updated GNB name and TAC in the `fiveg_gnb_identity` relation data bag."""
+    def _configure(self, _: EventBase) -> None:
+        """Publish gNB name `fiveg_core_gnb` relation data bag."""
         if not self.unit.is_leader():
             return
-        fiveg_gnb_identity_relations = self.model.relations.get(GNB_IDENTITY_RELATION_NAME)
-        if not fiveg_gnb_identity_relations:
-            logger.info("No %s relations found.", GNB_IDENTITY_RELATION_NAME)
-            return
+        if not self._relation_created(CORE_GNB_RELATION_NAME):
+            logger.info("No %s relations found.", CORE_GNB_RELATION_NAME)
```

Comment:
```suggestion
            logger.info("No %s relations found.", CORE_GNB_RELATION_NAME)
            return
```
We need a return here.